openmetadata-service/src/main/java/org/openmetadata/service/resources/search/ApplyOMDRulesForElasticSearch.java

```diff
@@ -0,0 +1,234 @@
+                || rule.getOperations().contains(MetadataOperation.VIEW_ALL))) {
+          // Check Conditions
+          if (rule.getCondition() == null) {
+            if (rule.getEffect().equals(Rule.Effect.DENY)) {
+              baseQueryBuilder.mustNot(searchSourceBuilder.query()); // 'MustNot' because of DENY rule
+            } else {
+              baseQueryBuilder.should(searchSourceBuilder.query()); // 'should' because of Allow rule
+            }
+          } else {
+            SpelExpression ruleExpression = (SpelExpression) rule.getExpression();
+            var rootAst = ruleExpression.getAST();
+            BoolQueryBuilder boolQueryBuilder = createQueryBuilderBasedOnRule(rootAst);
+            if (rule.getEffect().equals(Rule.Effect.DENY)) {
+              baseQueryBuilder.mustNot(boolQueryBuilder); // 'MustNot' because of DENY rule
+            } else {
+              baseQueryBuilder.should(boolQueryBuilder); // 'should' because of Allow rule
+            }
+          }
+          anyRuleApplied = true;
+        }
+      } // End of for
+    } // End of while
+
+    // If there is no rule applied to query, we return an empty result
+    if (!anyRuleApplied) {
+      return searchSourceBuilder.query(
+          QueryBuilders.boolQuery().must(searchSourceBuilder.query()).must(QueryBuilders.queryStringQuery("")));
+    }
+
+    /*
+        // For Apply just 'Deny' Rules
+        while (policies.hasNext()) {
+          SubjectContext.PolicyContext context = policies.next();
+          for (CompiledRule rule : context.getRules()) {
+            if ((rule.getResources().contains(indexToRuleMap.get(index))
+                    || rule.getResources().toString().toLowerCase().contains("all"))
+                    && (rule.getOperations().contains(MetadataOperation.ALL)
+                    || rule.getOperations().contains(MetadataOperation.VIEW_ALL))
+                    && rule.getEffect().equals(Rule.Effect.DENY)) {
+              // Check Conditions
+              if (rule.getCondition() == null) {
+                baseQueryBuilder.mustNot(searchSourceBuilder.query()); // 'MustNot' because of DENY rule
+              } else {
+                SpelExpression ruleExpression = (SpelExpression) rule.getExpression();
+                var rootAst = ruleExpression.getAST();
+                BoolQueryBuilder boolQueryBuilder = createQueryBuilderBasedOnRule(rootAst);
+                baseQueryBuilder.mustNot(boolQueryBuilder); // 'MustNot' because of DENY rule
+              }
+            }
+          } // End of for
+        } // End of while
+    */
+
+    // Combine searchSourceBuilder.query() with new Query that enforce policy rules
+    searchSourceBuilder.query(QueryBuilders.boolQuery().must(searchSourceBuilder.query()).must(baseQueryBuilder));
+    return searchSourceBuilder;
+  }
+
+  private BoolQueryBuilder createQueryBuilderBasedOnRule(SpelNode ast) throws IOException {
```

Comment:
IMO it would be better if we codify for each expression/rule to return the search condition for it. 
Example if the condition says isOwner() only allow, then we should return the condition.getSearchCondition(SecurityContext) -> return 'owner.name=%s'.format(SecurityContext.getUsername() or owner.name=(getTeams(SecurityContext.getUsername())